### ISSUE TITLE
Preventing the main html jQuery injection from loading too early

### ISSUE THREAD

**Issue by a13arnhe** [HGustavs/LenaSYS#2112]
Stop the jQuery from loading to early the html inserts, resulting in a blank page with an error like the one below where the innerHTML is not found.
![screen shot 2016-04-13 at 11 33 40](https://cloud.githubusercontent.com/assets/18159848/14489092/9cd1a90c-016b-11e6-9108-17299fff8241.png)

**Comment by HGustavs** [comment]
Use a delay of a few ms when the document has been created. if not found delay again a few times and if ot there after a few seconds make alert to contact admin.

**Comment by a13arnhe** [comment]
I have been thinking about your ideas since last night and I have few questions or concerns about it.

The javascript/jQuery code that is being run is being run at the top (head) of the main PHP files for each page instead of at the bottom, where it would be called after the DOM has been fully loaded. A way to fix this is either to put the `script` tags at the bottom of the main pages, or as the solution presented, which incases the actual search and insert of the DIV in a `document.ready()` function of jQuery.

Before:
`var slist=document.getElementById("accessedcontent");`
`slist.innerHTML = str;`
After:
`$(function(){`
`var slist=document.getElementById("accessedcontent");`
`slist.innerHTML=str;`
`});`

So if it would happen that the DOM would not load completely load and thus be "ready", then this code will never run, and never inserting the HTML that was fetched from the server via Ajax and PHP.

The reason why this solution was used is because the javascript is trying to load earlier than the actual page because of its placement, thus generating an error if the javascript is ahead of the DOM loading. It was not because it wasn't actually getting the data or such, just that the javascript is a tad bit faster in loading _(happened in 1 out of 10 loads maybe)_

Coming to the proposed solution of adding a few milliseconds of delay. There is really no need to delay the loading and then search for it again if not found, since the DOM will always be ready because of the added `document.ready()` solution. If the DOM won't be ready for some reason, then the problem does not really lie on this level and is more likely to be a PHP/MySQL problem instead, which this solution is not made for taking and and error checking. A previous issue **#480** has been closed and a solution presented to deal with if the config file for the database. Similar error handling as that could be used if there is a problem with the loading of the data itself in the PHP service of the current page of the system. Since also this javascript solution does not actually prevent "wrong" data to be inserted in case of an PHP/PDO error for example, fields could still be filled with _whitespace, 'undefined', or null_.

Again since this `.innerHTML` code will never run if the DOM won't be ready for some network reasons then invoking an alert box inside there would not really work as the jQuery will be waiting for the loading of the DOM until it can display the alert.

To sum up. Adding delay after the document is **known** to be ready I feel is not the best solution, and same with the alert box. Since most likely if there is an actual error with the page then it is most likely it will be server side DB or PHP related and not _race_ related to the DOM and the JS. Which then in turn should deal with the error before the insert even happens either by redirecting to an error page or injecting a pre-made error page into the HTML that will then be loading on DOM completion, letting the user know what is wrong. And keeping the solution as it is since the main problem was just the timing between and not the actual availability of the data itself.

**Comment by HGustavs** [comment]
I am not sure what accessed content does.

I would have put it into the render function that is called when the information is returned....
or
at a timer in that function if we are dependent on the browser finishing the assignment task.

If it is dependent on the generated content returned from the service it should not be called in the document at all.

**Comment by a13arnhe** [comment]
It is totally not depending on the generated content that is being returned from the service page for the current page. _Accessed_ is just a demo of part of the code, this `$(function(){});` has been added to all pages that put together a large string of text as the html page to be inserted and then finally inserted with this small `.innerHTML()`. And all those code snippets are in fact all in the **return** function that gets the data from the php/db. So for a solution just because the script is loaded above the main DOM I feel that this is the best and simplest solution for it. And "at a timer in that function if we are dependent on the browser finishing the assignment task.", is basically what `document.ready()` is doing, just based on the HTTP response code or the browser generally "telling" the jQuery that the page is ready.
